src/bot.ml

```
@@ -28,12 +28,52 @@ let bot_info : Bot_components.Bot_info.t =
   ; email= Config.bot_email toml_data
   ; domain= Config.bot_domain toml_data }
 
+let key = Config.github_private_key
+
+let app_id = Config.github_app_id toml_data
+
 let github_mapping, gitlab_mapping = Config.make_mappings_table toml_data
 
 let github_of_gitlab = Hashtbl.find gitlab_mapping
 
 let gitlab_of_github = Hashtbl.find github_mapping
 
+let action_as_github_app ~bot_info ~key ~app_id ~owner ~repo action =
+  GitHub_app.get_installation_token ~bot_info ~key ~app_id ~owner ~repo
+  >>= function
+  | Ok github_token ->
+      let bot_info = {bot_info with github_token} in
+      action ~bot_info
+  | Error _ ->
+      action ~bot_info
+
+let github_repo_of_gitlab_project_path gitlab_project_path =
+  let gitlab_full_name = gitlab_project_path in
+  let repo_full_name =
+    Option.value
+      (github_of_gitlab gitlab_full_name)
+      ~default:
+        ( Stdio.printf "No correspondence found for GitLab repository %s.\n"
+            gitlab_full_name ;
+          gitlab_full_name )
+  in
+  match Str.split (Str.regexp "/") repo_full_name with
+  | [owner_; repo_] ->
+      (owner_, repo_)
+  | _ ->
+      (* Shouldn't happen *)
+      ("owner", "repo")
```

In this case, you could simply raise an exception (with `failwith`).